ddos_guard.py

```diff
@@ -0,0 +1,73 @@
+"""
+Based on https://git.gay/a/ddos-guard-bypass/src/branch/master/index.js
+"""
+
+import logging
+import re
+import urllib.parse
+from http import HTTPStatus
+from http.cookies import SimpleCookie
+
+from aiohttp import ClientSession
+
+
+async def bypass(url: str, cookies=None, *, session: ClientSession):
+    if cookies is None:
+        headers_1 = {
+            "Accept": "text/html",
+            "Accept-Language": "en-US",
+            "Connection": "keep-alive",
+            "Sec-Fetch-Dest": "document",
+            "Sec-Fetch-Mode": "navigate",
+            "Sec-Fetch-Site": "none",
+            "Sec-Fetch-User": "?1",
+            "TE": "trailers",
+            "DNT": "1",
+        }
+        async with session.get(url, headers=headers_1) as response_1:
+            if response_1.status not in (HTTPStatus.OK, HTTPStatus.FORBIDDEN):
+                return
+            cookies = response_1.cookies
+
+    logging.info("[ddos_guard.bypass] Parsed cookies from %r: %r", url, cookies or None)
+
+    headers_2 = {
+        "Accept": "*/*",
+        "Accept-Language": "en-US,en;q=0.5",
+        "Accept-Encoding": "gzip, deflate",
+        "Referer": url,
+        "Sec-Fetch-Dest": "script",
+        "Sec-Fetch-Mode": "no-cors",
+        "Sec-Fetch-Site": "cross-site"
+    }
+    async with session.get("https://check.ddos-guard.net/check.js", cookies=cookies, headers=headers_2) as response_2:
+        text = await response_2.text()
+
+    match = re.search(r"'(?P<path>/.well-known/ddos-guard/id/\S+)'", text)
```

Comment:
There is a `etag` header in response, it contains id we need. Also this id available as `__ddg2` cookie value.

Reply:
Also, below we can live without "path" from regexp as it is standard one: "/.well-known/ddos-guard/id/{__ddg2}"